### ISSUE TITLE
Request to add "keys_to_ignore_at_inference" in the configuration_gpt_neox.py to resolve the failure in downstream repo

### ISSUE THREAD

**Issue by IT-Forrest** [huggingface/transformers#29376]
Compared with the configuration file of the LLama model, https://github.com/huggingface/transformers/blob/main/src/transformers/models/llama/configuration_llama.py#L115 
![image](https://github.com/huggingface/transformers/assets/9160073/944fd098-afac-401d-87f7-72f2339deb23)

the `keys_to_ignore_at_inference = ["past_key_values"]` is missed in the configuration file of the gpt-neox, 
https://github.com/huggingface/transformers/blob/0ad770c3733f9478a8d9d0bc18cc6143877b47a2/src/transformers/models/gpt_neox/configuration_gpt_neox.py#L107

Due to missing the attribute `keys_to_ignore_at_inference `, the **logits and logits dtypes mismatch** will be hit in the downstream repo, e.g. optimum-habana. The logits dtypes mismatch behaviors similarly in the issue https://github.com/huggingface/optimum-habana/pull/457 

Question: could we add "keys_to_ignore_at_inference" in the configuration_gpt_neox.py just as that in configuration_llama.py?

**Comment by ArthurZucker** [comment]
do you want to open a PR for a fix? 🤗

**Comment by IT-Forrest** [comment]
Hi @ArthurZucker, yes, hope there will be a PR for this "keys_to_ignore_at_inference" issue.
Since we don't have the interface person with transformers, how about you file the PR for us?
Or if you want me to file the PR, please kindly guide me the routine. Thanks a lot

**Comment by JJJYmmm** [comment]
Hello, Sorry to disturb you. Could you explain why `keys_to_ignore_at_inference` needs to add `past_key_values`? `past_key_value` is used to cache kv in auto-regressive inference, if we ignore it, how to do kv-cache?
